Update changes from tier1 machines for 1.8.0 release

RatkoVasic-NOAA opened the pull request:
### Summary

Update changes for tier1 machines for 1.8.0

### Testing

Install 1.8.0

### Applications affected

All

### Systems affected

Tier1

### Dependencies

### Issue(s) addressed

Part of issue #1278 

### Checklist
- [x] This PR addresses one issue/problem/enhancement, or has a very good reason for not doing so.
- [x] These changes have been tested on the affected systems and applications.
- [x] All dependency PRs/issues have been resolved and this PR can be merged.

RatkoVasic-NOAA committed:
start branch

RatkoVasic-NOAA committed:
Update hera mirrors.yaml

RatkoVasic-NOAA committed:
Update jet mirrors.yaml

Place mirror files to more friendly space

RatkoVasic-NOAA committed:
Remove gettext from jet packages.yaml

RatkoVasic-NOAA committed:
Set prefix for intel-oneapi-mpi Jet packages_intel.yaml

RatkoVasic-NOAA committed:
Add gcc to Gaea-C5 compilers.yaml

RatkoVasic-NOAA committed:
Add gcc to gaea-c6 compilers.yaml

RatkoVasic-NOAA committed:
Update noaa-cloud config files

RatkoVasic-NOAA committed:
Remove gettext from packages.yaml in noaa-aws

RatkoVasic-NOAA committed:
Update README.md file for noaa-aws 1.8.0 installation.

RatkoVasic-NOAA committed:
Checkout correct branch while cloning.

RatkoVasic-NOAA committed:
chmod 755 to 777

RatkoVasic-NOAA committed:
Merge branch 'release/1.8.0' into 1.8.0-fixes

AlexanderRichert-NOAA commented on configs/sites/tier1/hera/mirrors.yaml:
```diff
@@ -1,15 +1,15 @@
   local-source:
     fetch:
-      url: file:///contrib/spack-stack/mirror
+      url: file:///scratch1/NCEPDEV/nems/role.epic/spack-stack/mirror-1.8.0
```
Do the mirror paths need to be version-specific? Why not use one source cache across versions so we don't have to redownload for each release?

RatkoVasic-NOAA replied:
I thought mirror is containing particular versions of apps, which differ from release to release?

climbfuji replied:
mirrors are not version specific, should always point to the one authoritative mirror

AlexanderRichert-NOAA replied:
Yes, but cached packages are all differentiated by sha256 or commit hash, so I don't think there's any downside to putting them all in one big pot (and then we get the benefit of minimizing disk space and saving time by reusing those packages that don't change between versions). It's not a huge deal since in the future the source cache on each system will get populated by automated builds. Anyway, I'll approve the PR and it's up to you how you want to handle the source cache paths.

RatkoVasic-NOAA replied:
I'll make change.

climbfuji commented on configs/sites/tier1/noaa-aws/README.md:
```diff
@@ -1,164 +1,18 @@
-mkdir -p /contrib/admin
-cat <<EOF > /contrib/admin/basic_setup.sh
-#!/bin/bash
+## Steps to perform to install spack-stack version 1.8.0
```
what happened to all the instructions on the left - are they now baked into the default image? if not, they need to stay because somebody starting from a new image will need them

RatkoVasic-NOAA replied:
This is not needed anymore. All was because ecflow and it's dependences. On NOAA-Cloud spack-stack can now be installed from out-of-box.

climbfuji replied:
That's awesome, thanks for the clarification!

climbfuji replied:
As discussed, please bring the documentation back that is needed to install the qt@5 dependency for ecflow

RatkoVasic-NOAA replied:
I'll add part how to install qt5

climbfuji commented on configs/sites/tier1/noaa-aws/packages.yaml:
```diff
@@ -1,115 +1,114 @@
-    - spec: diffutils@3.3
+    - spec: diffutils@3.6
       prefix: /usr
   ecflow::
```
no external ecflow - but need external qt@5

RatkoVasic-NOAA replied:
That is now in common/modules.yaml and common/packages.yaml:
```
    qt:
      require: '@5'
```

climbfuji replied:
yes but you should configure the external qt5 package so that you don't need to build it (often doesn't work as expected). since you were able to build ecflow outside of spack, the necessary qt 5 libraries must be somewhere on the system

RatkoVasic-NOAA replied:
OK, I'm not sure how it worked. I intentionally didn't install:
```
#yum install -y qt5-qtbase-devel
#yum install -y qt5-qtsvg-devel
#yum install -y xorg-x11-xauth
#yum install -y xorg-x11-apps
#yum install -y perl-IPC-Cmd
#yum install -y gettext-devel
#yum install -y m4
```
Which we were supposed to install each time we start cloud instance.
Ecflow was installed using qt5 (which will be lost each time we logout).
Are you saying that we have to install qt5 again each time we login (like before)?
ecflow is still there, on permanent place (/contrib/spack-stack-rocky8/ecflow-5.8.4/)

As for m4, gettext, ... they were successfully installed by spack-satck-1.8.0.

climbfuji replied:
I think you need to install those packages every time, but you can easily create a bootstrap script and run that in your parallelworks config

RatkoVasic-NOAA replied:
That's OK, but why do we need to install them anyway, spack stack does not depend on them?

climbfuji replied:
qt5 does depend on them, and you need qt5 to build ecflow+ui - you can choose to let spack build all of that, not sure it works though

climbfuji commented on configs/sites/tier1/noaa-aws/packages.yaml:
```diff
@@ -1,115 +1,114 @@
-    - spec: pkg-config@0.27.1
+    - spec: pkg-config@1.4.2
       prefix: /usr
   rsync:
```
external qt@5 missing

RatkoVasic-NOAA replied:
That is now in common/modules.yaml and common/packages.yaml:
```
    qt:
      require: '@5'
```

RatkoVasic-NOAA committed:
Use old mirror site

RatkoVasic-NOAA committed:
Update documentation of how to install spack-stack on Tier 1 machines.

climbfuji commented on configs/sites/tier1/hera/mirrors.yaml:
```diff
@@ -1,15 +1,15 @@
   local-source:
     fetch:
-      url: file:///contrib/spack-stack/mirror
+      url: file:///scratch1/NCEPDEV/nems/role.epic/spack-stack/source-cache
```
please revert

RatkoVasic-NOAA replied:
I moved (large) mirror directory as suggested by @AlexanderRichert-NOAA , we are not supposed to store large datasets on /contrib space.

AlexanderRichert-NOAA replied:
The /contrib space is needed for performance/stability of the installed packages themselves so that model developers aren't held up by parallel filesystem issues, but we don't need that for the source/build caches (i.e., we're not in "real" trouble if the filesystem crashes for a day or two and we can't access the mirrors). The admins have already been fairly generous as far as disk usage/contrib policies/etc., so I think it would be best to keep the caches in "bulk storage" on the parallel filesystem. That said, I would suggest making the same change for Jet unless there's some reason not to.

RatkoVasic-NOAA replied:
I thought I did it for Jet as well, now I see I didn't. I'll do it over there as well.

climbfuji replied:
Thanks, that makes sense.

climbfuji commented on configs/sites/tier1/hera/mirrors.yaml:
```diff
@@ -1,15 +1,15 @@
       endpoint_url: null
     push:
-      url: file:///contrib/spack-stack/mirror
+      url: file:///scratch1/NCEPDEV/nems/role.epic/spack-stack/source-cache
```
please revert

RatkoVasic-NOAA replied:
Sabe as first remark.

climbfuji commented on doc/source/PreConfiguredSites.rst:
```diff
@@ -84,12 +86,10 @@ MSU Orion
 
    module purge
-   module use /work/noaa/epic/role-epic/spack-stack/orion/modulefiles-rocky9
+   No need to load modules, spack-stack-1.8.0 have no dependences
```
```suggestion
```

climbfuji commented on doc/source/PreConfiguredSites.rst:
```diff
@@ -100,14 +100,10 @@ MSU Hercules
-   module use /work/noaa/epic/role-epic/spack-stack/hercules/modulefiles
-   module load ecflow/5.8.4
-   module load git-lfs/3.1.2
+   No need to load modules, spack-stack-1.8.0 have no dependences
```
```suggestion
```
no git-lfs dependency???

RatkoVasic-NOAA replied:
Nope.

climbfuji commented on doc/source/PreConfiguredSites.rst:
```diff
@@ -251,17 +247,10 @@ NOAA Parallel Works (AWS, Azure, Gcloud)
-   module load cmake/3.27.2
-   module load ecflow/5.8.4
-   module load git-lfs/2.4.1
+   No need to load modules, spack-stack-1.8.0 have no dependences
```
really?

RatkoVasic-NOAA replied:
Yes.

climbfuji replied:
nice :-)

climbfuji commented on doc/source/PreConfiguredSites.rst:
```diff
@@ -272,18 +261,14 @@ NOAA RDHPCS Gaea C5
-
 .. code-block:: console
 
+   These modules should be loaded by default, if not load (swap) with:
```
```suggestion
   # These modules should be loaded by default, if not load (swap) with:
```

climbfuji commented on doc/source/PreConfiguredSites.rst:
```diff
@@ -304,18 +289,14 @@ NOAA RDHPCS Gaea C6
-
 .. code-block:: console
 
+   These modules should be loaded by default, if not load (swap) with:
```
```suggestion
   # These modules should be loaded by default, if not load (swap) with:
```

climbfuji commented on doc/source/PreConfiguredSites.rst:
```diff
@@ -336,14 +317,10 @@ NOAA RDHPCS Hera
-   module use /scratch1/NCEPDEV/nems/role.epic/modulefiles
-   module load miniconda3/4.12.0
-   module load ecflow/5.8.4
+   No need to load modules, spack-stack-1.8.0 have no dependences
```
```suggestion
```

climbfuji commented on doc/source/PreConfiguredSites.rst:
```diff
@@ -357,15 +334,10 @@ NOAA RDHPCS Jet
-   module load miniconda/3.9.12
-   module load ecflow/5.5.3
-   module use /lfs4/HFIP/hfv3gfs/role.epic/modulefiles
+   No need to load modules, spack-stack-1.8.0 have no dependences
```
```suggestion
```

RatkoVasic-NOAA committed:
Update doc/source/PreConfiguredSites.rst

Co-authored-by: Dom Heinzeller <dom.heinzeller@icloud.com>

RatkoVasic-NOAA committed:
Update doc/source/PreConfiguredSites.rst

Co-authored-by: Dom Heinzeller <dom.heinzeller@icloud.com>

RatkoVasic-NOAA committed:
Update doc/source/PreConfiguredSites.rst

Co-authored-by: Dom Heinzeller <dom.heinzeller@icloud.com>

RatkoVasic-NOAA committed:
Update doc/source/PreConfiguredSites.rst

Co-authored-by: Dom Heinzeller <dom.heinzeller@icloud.com>

RatkoVasic-NOAA committed:
Update doc/source/PreConfiguredSites.rst

Co-authored-by: Dom Heinzeller <dom.heinzeller@icloud.com>

climbfuji commented on doc/source/PreConfiguredSites.rst:
```diff
@@ -251,17 +246,10 @@ NOAA Parallel Works (AWS, Azure, Gcloud)
-   module load cmake/3.27.2
-   module load ecflow/5.8.4
-   module load git-lfs/2.4.1
+   No need to load modules, spack-stack-1.8.0 have no dependences
```
```suggestion
```

RatkoVasic-NOAA committed:
Update doc/source/PreConfiguredSites.rst

Co-authored-by: Dom Heinzeller <dom.heinzeller@icloud.com>

RatkoVasic-NOAA committed:
Update Jet mirrors path

RatkoVasic-NOAA committed:
Update mirrors.yaml

natalie-perlin commented on configs/sites/tier1/noaa-aws/README.md:
````diff
@@ -1,164 +1,18 @@
-### Create a mysql config for local R2D2 use (if applicable)
-```
-sudo su
-cat <<EOF > /contrib/admin/my.cnf
````
We may still need to have /contrib/admin/my.cnf saved. It may not need to be done every time a new space for spack stack is configured, i.e., a new /contrib space is prepared

RatkoVasic-NOAA committed:
Apply changes for NOAA-cloud machines for spack-stack-1.8.0

climbfuji commented on configs/sites/tier1/noaa-aws/README.md:
```diff
@@ -16,3 +34,47 @@ spack concretize 2>&1 | tee log.concretize
+mv mysql-8.0.31-linux-glibc2.17-x86_64-minimal/* ..
+rmdir mysql-8.0.31-linux-glibc2.17-x86_64-minimal
+
+ ecFlow:
```
Why ecflow? I thought we had discussed that we don't need this anymore. But ok, let's leave it in the readme, it won't get used as long as the site config has it removed from the external packages and the qt5 external package added instead (if it isn't there already).

climbfuji commented on doc/source/PreConfiguredSites.rst:
```diff
@@ -100,14 +99,10 @@ MSU Hercules
-   module use /work/noaa/epic/role-epic/spack-stack/hercules/modulefiles
-   module load ecflow/5.8.4
-   module load git-lfs/3.1.2
+   No need to load modules, spack-stack-1.8.0 have no dependences
```
```suggestion
   # No need to load modules, spack-stack-1.8.0 have no dependences
```

climbfuji committed:
Apply suggestions from code review

climbfuji reviewed (commented):
Instructions for building external ecflow packages are still in the README for all three sites. The site configs still have external ecflow packages, but no external qt5 packages. I'll wait with merging this PR until Wednesday noon but then it will go in as is and the release will be tagged.

climbfuji committed:
Merge branch 'release/1.8.0' into 1.8.0-fixes

RatkoVasic-NOAA commented:
> Instructions for building external ecflow packages are still in the README for all three sites. The site configs still have external ecflow packages, but no external qt5 packages. I'll wait with merging this PR until Wednesday noon but then it will go in as is and the release will be tagged.

@climbfuji I'm finishing NOAA-cloud. GCP is installed. Azure is slower. I'm going to add 2 more commits for these two machines (as we speak).

Commit:
Update config files for NOAA-GCP

RatkoVasic-NOAA committed:
Update config files for NOAA cloud

RatkoVasic-NOAA commented:
@climbfuji I'm done with NOAA cloud updates. Removed instructions for ecflow as well.

climbfuji commented:
The three `noaa-{aws,gcloud,azure}` site configs still have external `ecflow` packages and no external `qt5` packages. I am going to merge this PR nonetheless so that we can tag the release and be done with it.

RatkoVasic-NOAA commented:
> The three `noaa-{aws,gcloud,azure}` site configs still have external `ecflow` packages and no external `qt5` packages. I am going to merge this PR nonetheless so that we can tag the release and be done with it.

I did that in the meantime. I removed ecflow from packages.yaml, and added external qt5 to it. It was just too late to add it to the repository. BTW, same is with Hera, Jet, Gaeas. I already fixed some of them (Jet, Orion, Hercules) and rest will do tomorrow. Eventually, we can add those to release/1.8.0.

climbfuji commented:
Fine with me. It won't be in the tagged code, but at least in the release branch. Thanks @RatkoVasic-NOAA !
